File: JavaScript/CreditCardValidator.js
function getCardType(cardNumber) {
  if (cardNumber.startsWith("4")) {
    return "Visa";
  } else if (cardNumber.startsWith("5")) {
    return "MasterCard";
  } else if (cardNumber.startsWith("37")) {
    return "American Express";
  } else if (cardNumber.startsWith("6")) {
    return "Discover";
  } else {
    return "Unknown";
  }
}

function sumOfDoubleEvenPlace(cardNumber) {
  let sum = 0;
  for (let position = cardNumber.length - 2; position >= 0; position -= 2) {
    let digit = parseInt(cardNumber.charAt(position), 10);
    let doubledValue = digit * 2;
    sum += (doubledValue > 9) ? (doubledValue - 9) : doubledValue;
  }
  return sum;
}

function sumOfOddPlace(cardNumber) {
  let sum = 0;
  for (let position = cardNumber.length - 1; position >= 0; position -= 2) {
    let digit = parseInt(cardNumber.charAt(position), 10);
    sum += digit;
  }
  return sum;
}

function isValid(cardNumber) {
  let evenSum = sumOfDoubleEvenPlace(cardNumber);
  let oddSum = sumOfOddPlace(cardNumber);
  let totalSum = evenSum + oddSum;
  return totalSum % 10 === 0;
}

const cardNumber = "4388576018402626";
console.log("Card Type:", getCardType(cardNumber));
console.log("Is Valid:", isValid(cardNumber));

File: Python/Credit_Card_Validator.py
def get_card_type(card_number: str) -> str:
    if card_number.startswith("4"):
        return "Visa"
    elif card_number.startswith("5"):
        return "MasterCard"
    elif card_number.startswith("37"):
        return "American Express"
    elif card_number.startswith("6"):
        return "Discover"
    else:
        return "Unknown"


def sum_of_double_even_place(card_number: str) -> int:
    total_sum = 0
    for position in range(len(card_number) - 2, -1, -2):
        digit = int(card_number[position])
        doubled_value = digit * 2
        total_sum += doubled_value - 9 if doubled_value > 9 else doubled_value
    return total_sum


def sum_of_odd_place(card_number: str) -> int:
    total_sum = 0
    for position in range(len(card_number) - 1, -1, -2):
        digit = int(card_number[position])
        total_sum += digit
    return total_sum


def is_valid(card_number: str) -> bool:
    even_sum = sum_of_double_even_place(card_number)
    odd_sum = sum_of_odd_place(card_number)
    total_sum = even_sum + odd_sum
    return total_sum % 10 == 0


card_number = input("Enter a credit card number: ")
card_type = get_card_type(card_number)
valid = is_valid(card_number)

print("Card Type:", card_type)
if valid:
    print(f"Card Number {card_number} is VALID")
else:
    print(f"Card Number {card_number} is INVALID")
